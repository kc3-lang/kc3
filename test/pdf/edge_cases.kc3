quote require PDF
require PDF
quote PDF.parse_str("   123   ")
PDF.parse_str("   123   ")
quote PDF.parse_str("(Hello \\n World)")
PDF.parse_str("(Hello \\n World)")
quote PDF.parse_str("(Tab\\tTest)")
PDF.parse_str("(Tab\\tTest)")
quote PDF.parse_str("(Quote\\\"Test)")
PDF.parse_str("(Quote\\\"Test)")
quote PDF.parse_str("(Backslash\\\\Test)")
PDF.parse_str("(Backslash\\\\Test)")
quote PDF.parse_str("(\\101\\102\\103)")
PDF.parse_str("(\\101\\102\\103)")
quote PDF.parse_str("(outer \\(inner\\) text)")
PDF.parse_str("(outer \\(inner\\) text)")
quote PDF.parse_str("<41 42 43 44>")
PDF.parse_str("<41 42 43 44>")
quote PDF.parse_str("/VeryLongNameWithManyCharactersToTestParsing")
PDF.parse_str("/VeryLongNameWithManyCharactersToTestParsing")
quote PDF.parse_str("/Name#20With#20Spaces")
PDF.parse_str("/Name#20With#20Spaces")
quote PDF.parse_str("[ ]")
PDF.parse_str("[ ]")
quote PDF.parse_str("<<>>")
PDF.parse_str("<<>>")
quote PDF.parse_str("[true false null]")
PDF.parse_str("[true false null]")
quote PDF.parse_str("[[[[/nested]]]]")
PDF.parse_str("[[[[/nested]]]]")
quote PDF.parse_str("<< /A << /B << /C /D >> >> >>")
PDF.parse_str("<< /A << /B << /C /D >> >> >>")
quote PDF.parse_str("999999999999999999")
PDF.parse_str("999999999999999999")
quote PDF.parse_str("-0")
PDF.parse_str("-0")
quote PDF.parse_str("12345.678")
PDF.parse_str("12345.678")
quote PDF.parse_str("true")
PDF.parse_str("true")
quote PDF.parse_str("false")
PDF.parse_str("false")
quote PDF.parse_str("42 0 obj /Test endobj")
PDF.parse_str("42 0 obj /Test endobj")
quote PDF.parse_str("42 5 R")
PDF.parse_str("42 5 R")
quote PDF.parse_str("0 0 R/Name")
PDF.parse_str("0 0 R/Name")
quote PDF.parse_str("123 % This is a comment\n")
PDF.parse_str("123 % This is a comment\n")
quote PDF.parse_str("<</Length 306/Filter/FlateDecode>>")
PDF.parse_str("<</Length 306/Filter/FlateDecode>>")
quote PDF.parse_str("333195 0 obj<</Subtype/Type1C/Length 3/Filter/FlateDecode>>stream\r\n123\r\nendstream\r\nendobj")
PDF.parse_str("333195 0 obj<</Subtype/Type1C/Length 3/Filter/FlateDecode>>stream\r\n123\r\nendstream\r\nendobj")
