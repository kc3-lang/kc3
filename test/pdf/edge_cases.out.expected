require PDF
PDF
PDF.parse_str("   123   ")
123
PDF.parse_str("(Hello \\n World)")
"Hello \n World"
PDF.parse_str("(Tab\\tTest)")
"Tab\tTest"
PDF.parse_str("(Quote\\\"Test)")
"Quote\"Test"
PDF.parse_str("(Backslash\\\\Test)")
"Backslash\\Test"
PDF.parse_str("(\\101\\102\\103)")
"ABC"
PDF.parse_str("(outer \\(inner\\) text)")
"outer (inner) text"
PDF.parse_str("<41 42 43 44>")
"ABCD"
PDF.parse_str("/VeryLongNameWithManyCharactersToTestParsing")
VeryLongNameWithManyCharactersToTestParsing
PDF.parse_str("/Name#20With#20Spaces")
:"Name With Spaces"
PDF.parse_str("[ ]")
[void]
PDF.parse_str("<<>>")
%{}
PDF.parse_str("[true false null]")
[true, false, void]
PDF.parse_str("[[[[/nested]]]]")
[[[[nested]]]]
PDF.parse_str("<< /A << /B << /C /D >> >> >>")
%{A: %{B: %{C: D}}}
PDF.parse_str("999999999999999999")
999999999999999999
PDF.parse_str("-0")
0
PDF.parse_str("1.23")
(F64) 1.23
PDF.parse_str("true")
true
PDF.parse_str("false")
false
PDF.parse_str("42 0 obj /Test endobj")
{:indirect_object, 42, 0, Test}
PDF.parse_str("42 5 R")
{:indirect_object, 42, 5}
PDF.parse_str("123 % This is a comment\n")
123
